tests/examples/digital_fingerprinting/test_dfp_file_to_df.py

```diff
@@ -0,0 +1,358 @@
+                              'appshield',
+                              'snapshot-1',
+                              'threadlist_2022-01-30_10-26-01.670391.json')
+    file_specs = fsspec.open_files(input_file)
+    assert len(file_specs) == 1
+    yield fsspec.core.OpenFile(fs=file_specs.fs, path=file_specs[0].path)
+
+
+def test_single_object_to_dataframe(single_file_obj: fsspec.core.OpenFile):
+    from dfp.stages.dfp_file_to_df import _single_object_to_dataframe
+
+    schema = DataFrameInputSchema(
+        column_info=[CustomColumn(name='data', dtype=str, process_column_fn=lambda df: df['data'].to_list()[0])])
+    df = _single_object_to_dataframe(single_file_obj, schema, FileTypes.Auto, False, {})
+
+    assert df.columns == ['data']
+    with open(single_file_obj.path, encoding='UTF-8') as fh:
+        d = json.load(fh)
+        expected_data = d['data']
+
+    df['data'].to_list() == expected_data
+
+
+def test_single_object_to_dataframe_timeout():
+    from dfp.stages.dfp_file_to_df import _single_object_to_dataframe
+
+    input_glob = os.path.join(TEST_DIRS.tests_data_dir, 'appshield', 'snapshot-1', 'fake_wont_match*.json')
+    bad_file = fsspec.core.OpenFile(fs=fsspec.open_files(input_glob).fs, path='/tmp/fake/doesnt/exit.csv')
+
+    assert _single_object_to_dataframe(bad_file, DataFrameInputSchema(), FileTypes.CSV, False, {}) is None
+
+
+@pytest.mark.usefixtures("restore_environ")
+def test_constructor(config: Config):
+    from dfp.stages.dfp_file_to_df import DFPFileToDataFrameStage
+
+    # The user may have this already set, ensure it is undefined
+    os.environ.pop('MORPHEUS_FILE_DOWNLOAD_TYPE', None)
+
+    schema = DataFrameInputSchema()
+    stage = DFPFileToDataFrameStage(config,
+                                    schema,
+                                    filter_null=False,
+                                    file_type=FileTypes.PARQUET,
+                                    parser_kwargs={'test': 'this'},
+                                    cache_dir='/test/path/cache')
+
+    assert isinstance(stage, SinglePortStage)
+    assert isinstance(stage, PreallocatorMixin)
+    assert stage._schema is schema
+    assert stage._file_type == FileTypes.PARQUET
+    assert not stage._filter_null
+    assert stage._parser_kwargs == {'test': 'this'}
+    assert stage._cache_dir.startswith('/test/path/cache')
+    assert stage._dask_cluster is None
+    assert stage._downloader.download_method == "dask_thread"
+
+
+@pytest.mark.usefixtures("restore_environ")
+@pytest.mark.parametrize('dl_type', ["single_thread", "multiprocess", "multiprocessing", "dask", "dask_thread"])
+def test_constructor_download_type(config: Config, dl_type: str):
+    from dfp.stages.dfp_file_to_df import DFPFileToDataFrameStage
+
+    os.environ['MORPHEUS_FILE_DOWNLOAD_TYPE'] = dl_type
+    stage = DFPFileToDataFrameStage(config, DataFrameInputSchema())
+    assert stage._downloader.download_method == dl_type
```

Comment:
This is really testing the `Downloader` class and not `DFPFileToDataFrameStage`.